src/modules/hip/kernel/rain.hpp

```diff
@@ -0,0 +1,308 @@
+    rain_hip_compute(srcPtr1, &src1_f24.f8[0], &src2_f8, &dst_f24.f8[0], &alpha_f4);
+    rain_hip_compute(srcPtr1, &src1_f24.f8[1], &src2_f8, &dst_f24.f8[1], &alpha_f4);
+    rain_hip_compute(srcPtr1, &src1_f24.f8[2], &src2_f8, &dst_f24.f8[2], &alpha_f4);
+    rpp_hip_pack_float24_pln3_and_store24_pkd3(dstPtr + dstIdx, &dst_f24);
+}
+
+template <typename T>
+RppStatus hip_exec_rain_tensor(T *srcPtr1,
+                               RpptDescPtr srcDescPtr,
+                               T *dstPtr,
+                               RpptDescPtr dstDescPtr,
+                               Rpp32f rainPercentage,
+                               Rpp32u rainWidth,
+                               Rpp32u rainHeight,
+                               Rpp32s slant,
+                               Rpp32f *alpha,
+                               RpptROIPtr roiTensorPtrSrc,
+                               RpptRoiType roiType,
+                               rpp::Handle& handle)
+{
+    if (roiType == RpptRoiType::LTRB)
+        hip_exec_roi_converison_ltrb_to_xywh(roiTensorPtrSrc, handle);
+
+    Rpp32f rainPercent = rainPercentage * 0.004f;
+    Rpp32u numDrops = static_cast<Rpp32u>(rainPercent * srcDescPtr->h * srcDescPtr->w);
+    std::srand(std::time(0));
+    T *rainLayer = reinterpret_cast<T *>(handle.GetInitHandle()->mem.mcpu.scratchBufferHost);
+    T initValue = 0;
+    if constexpr (std::is_same<T, Rpp8s>::value)
+    {
+        initValue = static_cast<T>(0x81);   // 0x81 represents -127 in signed 8-bit integer(Rpp8s).
+    }
+    std::memset(rainLayer, initValue, srcDescPtr->strides.nStride * sizeof(T));
+    // Choose the rain intensity value based on the data type
+    T rainValue = std::is_same<T, Rpp8u>::value ? static_cast<T>(RAIN_INTENSITY_8U) :
+                  std::is_same<T, Rpp8s>::value ? static_cast<T>(RAIN_INTENSITY_8S) :
+                  static_cast<T>(RAIN_INTENSITY_FLOAT);
+    Rpp32f slantPerDropLength = static_cast<Rpp32f>(slant) / rainHeight;
+    for (Rpp32u i = 0; i < numDrops; i++)
+    {
+        Rpp32u xStart = rand() % (srcDescPtr->w - slant);
+        Rpp32u yStart = rand() % (srcDescPtr->h - rainHeight);
+        for (Rpp32u j = 0; j < rainHeight; j++)
+        {
+            Rpp32u x = xStart + j * slantPerDropLength;
+            Rpp32u y = yStart + j;
+
+            if (x >= 0 && x < srcDescPtr->w && y < srcDescPtr->h)
+            {
+                T *rainLayerTemp = rainLayer + y * srcDescPtr->w + x;
+                for (int k = 0; k < rainWidth; k++)
+                    *(rainLayerTemp + k * srcDescPtr->strides.wStride) = rainValue;
+            }
+        }
+    }
+
+    T *rainLayerHip = reinterpret_cast<T *>(handle.GetInitHandle()->mem.mgpu.scratchBufferHip.floatmem);
+    CHECK_RETURN_STATUS(hipMemcpyAsync(rainLayerHip, rainLayer, srcDescPtr->w * srcDescPtr->h * sizeof(T), hipMemcpyHostToDevice, handle.GetStream()));
+
+    int globalThreads_x = (dstDescPtr->strides.hStride + 7) >> 3;
```

Comment:
Lets change globalThreads_x as below
```suggestion
    int globalThreads_x = (dstDescPtr->w + 7) >> 3;
```

Reply:
Done